code/src/components/GameScreen.js

```diff
@@ -0,0 +1,67 @@
+import React from 'react'
+import { useSelector, useDispatch } from 'react-redux';
+import game, { generateGame } from 'reducers/game';
+
+import GameImages from './GameImages';
+
+const GameScreen = () => {
+  const step = useSelector((store) => store.game.currentStep);
+  const history = useSelector((store) => store.game.history);
+
+  const dispatch = useDispatch();
+
+  const onNextStep = (action) => dispatch(generateGame(action));
+  const onPreviousStep = () => dispatch(game.actions.setPreviousStep());
+
+  const directions = (action) => {
+    const moveString = `Move ${action.direction}`
+
+    if (action.direction === 'North') {
```

Comment:
Nice way to add direction-specific formatting. This works just fine, if you had any more directions I'd suggest swapping if-else out for a switch case.